### PR TITLE
Minor improvments and fixes

### PR DESCRIPTION
**Pull request by MaxGraey**

### CODE REVIEW

File: assembly/index.ts [hunk 1]
```diff
@@ -17,37 +17,33 @@ export function initState(seed: f64): void {
...
 // Random u64 using xoroshiro128starstar
 // 2018 by David Blackman and Sebastiano Vigna (public domain)
 // http://xoshiro.di.unimi.it/xoroshiro128starstar.c
 @inline
 function nextUint64(): u64 {
-  let result: u64 = rotl<u64>(s0 * 5, 7) * 9;
+  let _s0 = s0;
+  let _s1 = s1;
+
+  let t = rotl((_s0 << 2) + _s0, 7);
```
Comment (krisselden) [inline assembly/index.ts]: Please make this a separate PR, I don't know if this handles overflow the same as the original and I don't have tests yet to verify it behaves the same as the original, the only thing I have is to visually check the benchmark looks uniformly distributed at the end.

**Review by krisselden — CHANGES_REQUESTED** [review]
This PR contains too many things in one PR. I'm ok with the miscellaneous cleanup, I'm good with the change to how the u64 gets mapped onto the fraction part f64 and reinterpreted, but I need to verify more about the change to the algorithm itself.

File: assembly/index.ts (hunk 1 above)
Comment (MaxGraey) [reply assembly/index.ts]: Sure. done

Comment (MaxGraey) [reply assembly/index.ts]: But transform `x * 5` to `(x << 2) + x` is safe operation and produce same overflow behaviour. The reason of doing this is most VMs can't optimize `mul/div/mod` by constant for 64-bit registers yet. You could see how this doing in GCC: https://godbolt.org/z/d15Ncm

Comment (krisselden) [reply assembly/index.ts]: Breaking up a PR was a suggestion because I would have merged the other parts first while clarifying the above.  You've clarified it well enough that I'm good with it now.  Have you thought about adding the optimization to binaryen which assemblyscript uses to optimize?

**Review by krisselden — APPROVED** [review]

File: assembly/index.ts (hunk 1 above)
Comment (MaxGraey) [reply assembly/index.ts]: If this concerns arithmetic for constants then it is outscope of binaryen. Regarding optimization with cached global variables then it will be to some extent in the next release. Actually both optimization expand code size which not preferable by binaryen. In the future VMs learn how to do it yourself, but this improvement process is not fast

**Comment by krisselden** [comment]
Thank you for this, I’ll merge and release when I get back in front of a computer.
